tuning/data/data_processors.py

```diff
@@ -393,23 +393,10 @@ def process_dataset_configs(
         self, dataset_configs: List[DataSetConfig], **kwargs
     ) -> Union[Dataset, IterableDataset]:
         train_dataset = None
+        state = PartialState()
 
-        if torch.distributed.is_available() and torch.distributed.is_initialized():
-            if torch.distributed.get_rank() == 0:
-                logger.info("Processing data on rank 0...")
-                train_dataset = self._process_dataset_configs(dataset_configs, **kwargs)
-            else:
-                train_dataset = None
-
-            # Use broadcast_object_list to share the dataset object across ranks
-            # TODO: Check if torch.distributed.barrier() is called in broadcast_object_list()
-            # See https://github.com/pytorch/pytorch/issues/56142
-            # for why the list is shared like this
-            to_share = [train_dataset]
-            torch.distributed.broadcast_object_list(to_share, src=0)
-            train_dataset = to_share[0]
-        else:
-            logger.info("Processing data...")
+        # The local_main_process_first context ensures that the main process runs first per node
```

Comment:
Please expand the comment here to include our rationale of using local main process first as we want to reuse HF cache and not redo computation all the time.